src/backends/perf_pt/collect.c

```diff
@@ -127,8 +128,28 @@ struct tracer_thread_args {
                         *err;               // Errors generated inside the thread.
 };
 
+// A data buffer sample indicating that new data is available in the AUX
+// buffer. This struct is not defined in a perf header, so we have to define it
+// ourselves.
+struct perf_record_aux_sample {
+    struct perf_event_header header;
+    __u64    aux_offset;
+    __u64    aux_size;
+    __u64    flags;
+    // ...
+    // More variable-sized data follows, but we don't use it.
+};
+
+// The data structure which is read(2) out of the Perf file descriptor.
```

Comment:
This is a weird sentence -- "reading out of a file descriptor" sounds odd. Also, is this the *only* thing we ever read from perf? Does everything else just get dumped into a memory buffer?

Reply:
It's the only thing ever read using the `read` system call, yes. But as you have said, there are two `mmap` regions which are being filled. What we read out of the fd tells us how/when to look at these buffers. It's complicated.

I'm not sure what's wrong with "reading out of a file descriptor" though. Do you prefer "reading from a file descriptor"?

For comparison, here's how the `perf_event_open` (dryly) uses prose for this:
> This field specifies the format of the data returned by read(2) on a perf_event_open() file descriptor.

Reply:
Maybe we could say something like "The data-structure returned via the perf file descriptor, which then allows us to make sense of the memory buffers"?

Reply:
I don't like "returned" here. I can think of a couple of ways to get data from an fd, and not just via  `read(2)`. E.g. `ioctl(2)`.

Reply:
I don't think you can read data via `ioctl`?

Anyway, maybe "The data-structure read via the perf file ..."?

Reply:
It depends on the ioctl in question, but you certainly can get stuff out:
```
An ioctl request has encoded in it whether the argument is an “in” parameter or “out” parameter
```

> "The data-structure read via the perf file ..."?

WFM